src/test/java/org/elasticsearch/test/rest/ElasticsearchRestTestCase.java

```diff
@@ -247,7 +247,11 @@ protected int maximumNumberOfShards() {
 
     @Override
     protected int maximumNumberOfReplicas() {
-        return 1; // never go crazy in the REST tests
+        if (cluster().numDataNodes() > 1) {
+            return 1;
+        } else {
+            return 0;
```

Comment:
I think we should always return 1 here. REST tests should run with default number of replicas (1) even against a single node, because that's what clients tests do too. We can never expect green unless we manually set number of replicas to 0 in a test, and we should align to clients builds here I think to prevent failures that don't repro for us. Thoughts?

BTW running against a single node is very helpful because it allows us to realize when we make the wrong assumption in a REST test, which in the past was only going to fail for clients and never for us.